dali/base/dasess.cpp

```diff
@@ -2068,16 +2068,13 @@ class CUserDescriptor: implements IUserDescriptor, public CInterface
     {
         if (mb.remaining() > 0)
         {
-            size32_t len = 0;
-            mb.read(len);
-            if (len)
-                sessionToken.append(len, mb.readDirect(len));
-
+            mb.read((unsigned &)sessionToken);
             if (mb.remaining() > 0)
             {
+                unsigned len = 0;
                 mb.read(len);
                 if (len)
-                    signature.append(len, mb.readDirect(len));
+                    signature.append(len, (const char *)mb.readDirect(len));
```

Comment:
> There is no set function that allows you to specify the length

MemoryBuffer::setBuffer

Reply:
Jake, I am appending this value from the input MemoryBuffer onto the tail end of another MemoryBuffer, so setBuffer will not work